astropy/time/tests/test_basic.py

```diff
@@ -1366,3 +1366,122 @@ class _Time(Time):
 
     assert t2.__class__ == _Time
     assert t1 == t2
+
+
+def test_ymdhms_init():
+    time = Time('2001-01-01T00:00:00')
+
+    time_dict = {
+        'year': 2001,
+    }
+
+    t1 = Time(time_dict, format='ymdhms')
+    assert t1 == time
+
+    time_dict['month'] = 1
```

Comment:
These test values are the same as the defaults, so you aren't really testing much here.  I.e. the code to use the input values might not be running at all but this test would not show it.

Reply:
Will change these. I used the default for ease of testing.